--- mkprodct.csh.orig	2007-04-07 09:45:01.000000000 +1000
+++ mkprodct.csh	2007-04-07 09:49:32.000000000 +1000
@@ -145,6 +145,24 @@
    
 endif
 
+#
+#  Choose your linker.
+#
+if ( $?TKLINKER ) then
+
+   echo " "
+   echo "      Using linker: "
+   echo "      $TKLINKER"
+
+else
+
+   set TKLINKER  =  "ld"
+   echo " "
+   echo "      Setting default linker:"
+   echo $TKLINKER
+   
+endif
+
 
 #
 #  What compile options do we want to use? If they were 
@@ -188,7 +206,7 @@
 #   Determine a provisional LIBRARY name.
 #
    foreach item ( `pwd` )
-      set LIBRARY = "../../lib/"$item:t
+      set LIBRARY = "../../lib/lib"$item:t
    end
 
 #
@@ -225,6 +243,7 @@
    echo "      Inserting objects in the library $LIBRARY ..."
    ar  crv $LIBRARY.a *.o
    ranlib  $LIBRARY.a
+   $TKLINKER -shared -o $LIBRARY.so --whole-archive $LIBRARY.a
    \rm                *.o    
    echo " "
 
