--- wscript.orig	2012-02-26 10:14:39.443670251 +1100
+++ wscript	2012-02-26 10:20:33.488244086 +1100
@@ -29,12 +29,3 @@
     # the gsettings schemas - TODO - use whatever default tools there are
     bld.install_files('${DATADIR}/glib-2.0/schemas',
                       'hamster-applet.gschema.xml')
-
-
-    def post(ctx):
-        if bld.is_install:
-            print "Compiling schema"
-            ctx.exec_command("glib-compile-schemas '%s'" % os.path.join(ctx.env['DATADIR'], "glib-2.0", "schemas"))
-
-
-    bld.add_post_fun(post)
