--- maxdim.h.orig	2007-06-27 11:59:59.666225710 +1000
+++ maxdim.h	2007-06-27 12:00:08.164929178 +1000
@@ -1,7 +1,7 @@
 c=======================================================================
 c - MAXBUF tells us how many words of memory we can use for data
 	INTEGER   MAXBUF
-        PARAMETER(MAXBUF=1000000)
+        PARAMETER(MAXBUF=2000000)
 c-----------------------------------------------------------------------
 c - MAXDIM is an often used parameter, to indicate maximum size of maps
 	INTEGER   MAXDIM
