--- configure.ac.old	2008-02-07 19:58:03.000000000 +1100
+++ configure.ac	2008-02-07 20:49:21.000000000 +1100
@@ -236,15 +236,13 @@
 AC_SUBST([X11LIBS])
 
 
-# Search for RPFITSLIB, we need the full pathname.
 AC_MSG_NOTICE([
 
-Looking for the full pathname of the RPFITS library...])
+Looking for RPFITS and related libraries...])
 
-for EXTDIR in $EXTDIRS ; do
-  AC_CHECK_FILES([$EXTDIR/librpfits.$SO $EXTDIR/librpfits.a],
-                 [RPFITSLIB="$EXTDIR/librpfits.a" ; break])
-done
+AC_LANG(Fortran 77)
+AC_CHECK_LIB([rpfits],  [rpfitsin],     [RPFITSLIB="-lrpfits $RPFITSLIB"],
+             [RPFITSLIB=], [$RPFITSLIB $X11LIB $XTRALIBS])
 
 if test "x$RPFITSLIB" = x ; then
   AC_MSG_WARN([RPFITS library not found.  Some tasks will fail to link.])
