--- rpfits.orig/GNUmakefile	2006-07-17 17:05:47.000000000 +1000
+++ rpfits/GNUmakefile	2007-01-11 22:38:49.000000000 +1100
@@ -74,8 +74,7 @@
 	-@ head -29 ./GNUmakefile
 	-@ echo ""
 else
-all : lib rpfhdr rpfex
-	-@ $(RM) *.o
+all : lib shared_lib rpfhdr rpfex 
 
 lib : $(RPFITSLIB) ;
 
@@ -147,3 +146,6 @@
 	-@echo ""
 	-@echo "MEMBERS =$(MEMBERS)"
 endif
+
+shared_lib : $(MEMBERS)
+	$(FC) -shared ${LDFLAGS} -Wl,-soname,librpfits.so.2 -o librpfits.so.2 ${FOBJS} ${COBJS}
