--- source/rltiles/Makefile.orig	2010-07-31 15:57:42.765413128 +1000
+++ source/rltiles/Makefile	2010-07-31 15:57:59.905411522 +1000
@@ -23,20 +23,20 @@
 LDFLAGS += -ldl -lpthread
 endif
 
-# Attempt to use a full compiler name, to make
-# distcc builds work nicely.
-LMACH := $(shell gcc -dumpmachine)-
-ifeq ($(LMACH),-)
-LMACH :=
-endif
-ifeq ($(shell which $(LMACH)gcc > /dev/null 2> /dev/null && echo "Yes"),)
-LMACH :=
-endif
-CXX = $(LMACH)g++
-ifdef ARCH
-CF_ARCHS := $(patsubst %,-arch %,$(ARCH))
-CXX += $(CF_ARCHS)
-endif
+# # Attempt to use a full compiler name, to make
+# # distcc builds work nicely.
+# LMACH := $(shell gcc -dumpmachine)-
+# ifeq ($(LMACH),-)
+# LMACH :=
+# endif
+# ifeq ($(shell which $(LMACH)gcc > /dev/null 2> /dev/null && echo "Yes"),)
+# LMACH :=
+# endif
+# CXX = $(LMACH)g++
+# ifdef ARCH
+# CF_ARCHS := $(patsubst %,-arch %,$(ARCH))
+# CXX += $(CF_ARCHS)
+# endif
 
 DELETE = rm -f
 
