--- CMakeLists.txt.orig	2008-03-19 21:55:33.000000000 +1100
+++ CMakeLists.txt	2008-03-19 21:55:47.000000000 +1100
@@ -187,7 +187,7 @@
 endif(HAVE_QHULL)
 
 if(WITH_FREETYPE)
-  set_source_files_properties(plfreetype.c PROPERTIES COMPILE_FLAGS "-I${FREETYPE_INCLUDE_DIR}")
+  set_source_files_properties(plfreetype.c PROPERTIES COMPILE_FLAGS "${FREETYPE_INCLUDE_DIR}")
   set(
   libplplot${LIB_TAG}_LINK_LIBRARIES
   ${libplplot${LIB_TAG}_LINK_LIBRARIES}
