--- Slim/Web/HTTP.pm.old	2010-04-08 22:49:36.000000000 +0100
+++ Slim/Web/HTTP.pm	2010-04-08 22:49:36.000000000 +0100
@@ -2199,9 +2199,10 @@
 		# only kill the timer if we were called for the active streaming connection;
 		# otherwise we might kill the timer related to the next connection too.
 		Slim::Utils::Timers::killTimers($client, \&tryStreamingLater);
-		
-		Slim::Networking::Select::addWrite($httpClient, \&sendStreamingResponse, 1);
 	}
+
+	# Bug 14740 - always call sendStreamingResponse so we ensure the socket gets closed
+	Slim::Networking::Select::addWrite($httpClient, \&sendStreamingResponse, 1);
 }
 
 sub forgetClient {
