--- Makefile.in.orig	2007-11-29 19:51:42.000000000 +1100
+++ Makefile.in	2007-11-29 19:52:03.000000000 +1100
@@ -75,7 +75,7 @@
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $? $(AM_LDFLAGS) $(LDFLAGS) -o $@
 DIST_SOURCES = wndo-ctrl.c
 ETAGS = etags
 CTAGS = ctags
@@ -234,7 +234,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 wndo-ctrl$(EXEEXT): $(wndo_ctrl_OBJECTS) $(wndo_ctrl_DEPENDENCIES) 
 	@rm -f wndo-ctrl$(EXEEXT)
-	$(LINK) $(wndo_ctrl_LDFLAGS) $(wndo_ctrl_OBJECTS) $(wndo_ctrl_LDADD) $(LIBS)
+	$(LINK) $(wndo_ctrl_LDFLAGS) $(wndo_ctrl_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
