diff -Nur main.C.orig main.C
--- main.C.orig
+++ main.C
@@ -210,7 +210,7 @@
     char buffer[100], *p;
     char *(*codeconv)(const char*);
 
-    strcpy(buffer, operation);
+    strncpy(buffer, operation, sizeof(buffer)-1);
     if (strchr(buffer, '-') == NULL) {
         strcat(buffer, "-" KANJICODE);
     }
@@ -280,6 +280,7 @@
     int          num_piece_x, num_piece_y;
     Widget       topform, base;
     XrmDatabase  rdb1, rdb2;
+    char         debscoredir[] = "/var/games";
 
     setlocale(LC_ALL, "");
     XtSetLanguageProc(NULL, NULL, NULL);
@@ -342,7 +343,8 @@
                   NULL);
 
     if ((dat_directory = getenv("XSHISENDAT")) == NULL)
-        dat_directory = globRes.scoreDir;
+          dat_directory = debscoredir;
+/*        dat_directory = globRes.scoreDir; */
 
     sc = new Score(toplevel);
     scorefile = new char[strlen(dat_directory)+strlen(globRes.scoreFile)+1];
@@ -351,6 +353,7 @@
     delete[] scorefile;
 
     if (globRes.scoreOnly) {
+#ifndef NO_GLOBAL_HIGHSCORE
         sc->DisplayScore(initial_game_state);
 #if USE_MOTIF
         XtAddCallback(*sc, XmNokCallback, (XtCallbackProc)ExitCB, NULL);
@@ -358,6 +361,10 @@
         XtAddCallback(XtNameToWidget(*sc, "*ok_button"),
                       XtNcallback, (XtCallbackProc)ExitCB, NULL);
 #endif
+#else /* NO_GLOBAL_HIGHSCORE */
+	fprintf(stderr, "System score files not enabled.\n");
+	exit(1);
+#endif /* NO_GLOBAL_HIGHSCORE */
     }
     else {
         GetGameSize(initial_game_state, num_piece_x, num_piece_y);
diff -Nur .orig/menubar.C /menubar.C
--- menubar.C.orig
+++ menubar.C
@@ -158,8 +158,13 @@
                   (XtCallbackProc)GameCB, (XtPointer)1);
     XtAddCallback(XtNameToWidget(menushell0, "button_2"), XtNcallback,
                   (XtCallbackProc)GameCB, (XtPointer)2);
+#ifdef NO_GLOBAL_HIGHSCORE
+    XtVaSetValues(XtNameToWidget(menushell0, "button_3"), XtNsensitive,
+                  False, NULL);
+#else
     XtAddCallback(XtNameToWidget(menushell0, "button_3"), XtNcallback,
                   (XtCallbackProc)GameCB, (XtPointer)3);
+#endif
     XtAddCallback(XtNameToWidget(menushell0, "button_4"), XtNcallback,
                   (XtCallbackProc)GameCB, (XtPointer)4);
     XtAddCallback(XtNameToWidget(menushell0, "button_5"), XtNcallback,
diff -Nur .orig/score.C /score.C
--- score.C.orig
+++ score.C
@@ -372,12 +372,15 @@
     time_t t;
     struct tm *tp;
     struct passwd *pw;
-    char   namebuf[128], myname[NAMELEN+1], gecos[128], *po;
+    char   namebuf[142], myname[NAMELEN+1], gecos[128], *po;
 
+#ifndef NO_GLOBAL_HIGHSCORE
+    
     s1 = scoreToRegister / 1000;
     ms_to_hms(scoreToRegister, h, m, s);
     pw = getpwuid(getuid());
-    strcpy(gecos, pw->pw_gecos);
+    strncpy(gecos, pw->pw_gecos, sizeof(gecos));
+    gecos[sizeof(gecos) - 1] = '\0';
     if ((po = strchr(gecos, ',')) != NULL)
         *po = 0;
     sprintf(namebuf, "%-8.8s (%s)", pw->pw_name, gecos);
@@ -424,6 +427,7 @@
     sprintf(rec[i].time, "%2.2d:%2.2d:%2.2d", tp->tm_hour, tp->tm_min, tp->tm_sec);
     if (writefile() == 0)
         DisplayScore(game);
+#endif /* NO_GLOBAL_HIGHSCORE */
 }
 
 void
@@ -666,9 +670,9 @@
         bp += strlen(bp);
     }
     if (rcount[0]) {
-        x = rtime[0] / rcount[0];
-        y = (double)rlev[0] / (double)rcount[0];
-        ms_to_hms(x, t1, t2, t3);
+	x = rtime[0] / rcount[0];
+	y = (double)rlev[0] / (double)rcount[0];
+	ms_to_hms(x, t1, t2, t3);
         bp += strlen(bp);
         sprintf(bp, "    %-14.14s: %2.2d:%2.2d:%2.2d (%4.1f) %6d %s\n",
                 res_strings[8], t1, t2, t3, y, rcount[0], res_strings[11]);
diff -Nur .orig/readxpm.C /readxpm.C
--- readxpm.C.orig
+++ readxpm.C
@@ -12,7 +12,7 @@
   char buffer[1024];
 
   for(int i=0; i<PKIND; i++) {
-    sprintf(buffer, "%s/%s/%s.xpm", directory, subdir, files[i]);
+    snprintf(buffer, sizeof(buffer)-1, "%s/%s/%s.xpm", directory, subdir, files[i]);
     Mp[i].ReadFile(w, buffer, i, globRes.colorCloseness);
   }
 }
diff -Nur .orig/Makefile.in /Makefile.in
--- Makefile.in.orig
+++ Makefile.in
@@ -17,7 +17,7 @@
 
 XSHISENLIB = @datadir@/xshisen
 XSHISENDAT = @sharedstatedir@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = @CXXFLAGS@ -DNO_GLOBAL_HIGHSCORE
 CPPFLAGS = $(CPPOPTS) @CPPFLAGS@
 CPPOPTS  =
 LDFLAGS  = @LDFLAGS@
@@ -57,14 +57,15 @@
 	@chmod 755 $(bindir)
 	$(INSTALLP) $(PROGS) $(bindir)/$(PROGS)
 	@echo "XShisen data directory is " $(XSHISENLIB)
-	@$(MKDIR) $(XSHISENDAT)
-	@chmod 755 $(XSHISENDAT)
+	@echo "XShisen higscore/shared data directory is " $(XSHISENDAT)
+	#@$(MKDIR) $(XSHISENDAT)
+	#@chmod 755 $(XSHISENDAT)
 	@if [ -f $(XSHISENDAT)/$(SCF) ]; then \
 	echo "Old high-score file exists."; \
-	else\
+	else \
 	$(INSTALL) -m 0666 $(SCF) $(XSHISENDAT)/$(SCF); \
 	fi
-	@$(MKDIR) $(XSHISENLIB)
+	## @$(MKDIR) $(XSHISENLIB)
 	@set -x; \
 	for dir in pixmaps/s?; do \
 		dn=`basename $$dir`; \
@@ -76,6 +77,7 @@
 	done
 
 install-resource: $(AD)
+	echo destdir $(DESTDIR)
 	$(MKDIR) $(DESTDIR)$(XAD_DIR)/app-defaults
 	$(INSTALLD) $(AD) $(DESTDIR)$(XAD_DIR)/app-defaults/$(class_name)
 	-[ -n $(LANG) ] && $(MKDIR) $(DESTDIR)$(XAD_DIR)/$(LANG)/app-defaults
@@ -105,11 +107,12 @@
 
 $(AD): $(exec_name) lang/XShisen.ad
 	-$(MKDIR) work
-	$(CODECONV) < lang/XShisen.ad > $(AD)
-	-@set -x; \
-	for lang in $(LANGS); do \
-	$(CODECONV) < lang/XShisen.ad.$$lang > $(AD).$$lang; \
-	done
+	cp lang/* work
+	#$(CODECONV) < lang/XShisen.ad > $(AD)
+	#-@set -x; \
+	#for lang in $(LANGS); do \
+	#$(CODECONV) < lang/XShisen.ad.$$lang > $(AD).$$lang; \
+	#done
 
 .SUFFIXES: .C
 .C.o:
diff -Nur .orig/lang/XShisen.ad /lang/XShisen.ad
--- lang/XShisen.ad.orig
+++ lang/XShisen.ad
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: Shisen-sho for X11 ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: Shisen-sho for X11 ver 1.50\n\n\
  Shisen-sho for X11    by Masaoki Kobayashi <masaoki@techfirm.co.jp>\n\
    Shisen-sho for Win  by Shinji Ono <ono@urban.ne.jp>\n\
    Picture             by Takayuki Akiyama <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja_JP.eucJP /lang/XShisen.ad.ja_JP.eucJP
--- lang/XShisen.ad.ja_JP.eucJP.orig
+++ lang/XShisen.ad.ja_JP.eucJP
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: ªÕ¿Óæ  for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: ªÕ¿Óæ  for X11 Ver 1.50\n\n\
  ªÕ¿Óæ  for X11     by æÆŒ” ¿µ∂Ω <masaoki@techfirm.co.jp>\n\
    ªÕ¿Óæ  for Win   by æÆÃÓ ø∏∆Û <ono@urban.ne.jp>\n\
    «◊•«•∂•§•Û       by Ω©ª≥ Œ¥«∑ <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja_JP /lang/XShisen.ad.ja_JP
--- lang/XShisen.ad.ja_JP.orig
+++ lang/XShisen.ad.ja_JP
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: ªÕ¿Óæ  for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: ªÕ¿Óæ  for X11 Ver 1.50\n\n\
  ªÕ¿Óæ  for X11     by æÆŒ” ¿µ∂Ω <masaoki@techfirm.co.jp>\n\
    ªÕ¿Óæ  for Win   by æÆÃÓ ø∏∆Û <ono@urban.ne.jp>\n\
    «◊•«•∂•§•Û       by Ω©ª≥ Œ¥«∑ <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja /lang/XShisen.ad.ja
--- lang/XShisen.ad.ja.orig
+++ lang/XShisen.ad.ja
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: ªÕ¿Óæ  for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: ªÕ¿Óæ  for X11 Ver 1.50\n\n\
  ªÕ¿Óæ  for X11     by æÆŒ” ¿µ∂Ω <masaoki@techfirm.co.jp>\n\
    ªÕ¿Óæ  for Win   by æÆÃÓ ø∏∆Û <ono@urban.ne.jp>\n\
    «◊•«•∂•§•Û       by Ω©ª≥ Œ¥«∑ <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja_JP.SJIS /lang/XShisen.ad.ja_JP.SJIS
--- lang/XShisen.ad.ja_JP.SJIS.orig
+++ lang/XShisen.ad.ja_JP.SJIS
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: élêÏè» for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: élêÏè» for X11 Ver 1.50\n\n\
  élêÏè» for X11     by è¨ó— ê≥ãª <masaoki@techfirm.co.jp>\n\
    élêÏè» for Win   by è¨ñÏ êWìÒ <ono@urban.ne.jp>\n\
    îvÉfÉUÉCÉì       by èHéR ó≤îV <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.pl /lang/XShisen.ad.pl
--- lang/XShisen.ad.pl.orig
+++ lang/XShisen.ad.pl
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: Shisen-sho dla X11 wersja XSHISEN_VERSION\n\n\
+XShisen*aboutString: Shisen-sho dla X11 wersja 1.50\n\n\
  Shisen-sho dla X11    autorstwa Masaoki Kobayashi <masaoki@techfirm.co.jp>\n\
    Shisen-sho dla Win  autorstwa Shinji Ono <ono@urban.ne.jp>\n\
    Grafika(kafle)      autorstwa Takayuki Akiyama <taco3@ybb.ne.jp>
@@ -14,7 +14,7 @@
 XShisen*score*strPlayed: Rozegranych
 XShisen*score*strCompleted: UkoÒczonych
 XShisen*score*strTedumari: Zablokowanych
-XShisen*score*strSearched: Search used
+XShisen*score*strSearched: Podpowiedzianych
 XShisen*score*strGames: gier
 XShisen*score*strAverage: ¶rednio
 XShisen*score*strTotal: Wszystkich
@@ -46,7 +46,7 @@
 XShisen*mb.button_1.mnemonic: O
 XShisen*mb.button_2.labelString: Konfiguracja
 XShisen*mb.button_2.mnemonic: C
-XShisen*mb.button_3.labelString: Picture
+XShisen*mb.button_3.labelString: Kafle
 XShisen*mb.button_3.mnemonic: P
 XShisen*pd0.button_0.labelString: Start
 XShisen*pd0.button_0.mnemonic: D
@@ -87,10 +87,10 @@
 XShisen*pd2.button_0.labelString: Normalne   (18x 8)
 XShisen*pd2.button_1.labelString: Duøe       (24x12)
 XShisen*pd2.button_2.labelString: Ogromne    (36x16)
-XShisen*pd2.button_3.labelString: Click Trial
+XShisen*pd2.button_3.labelString: Szybkie klikanie
 XShisen*pd2.button_4.labelString: Grawitacja
-XShisen*pd3.button_0.labelString: Picture#1
-XShisen*pd3.button_1.labelString: Picture#2
+XShisen*pd3.button_0.labelString: Zestaw 1
+XShisen*pd3.button_1.labelString: Zestaw 2
 XShisen*drawing.translations: #replace <Btn1Down>: PickupPiece()\n\
         <Btn2Down>(2): menu(0,0)\n\
         <Btn3Down>: CancelPiece()
@@ -118,7 +118,7 @@
 XShisen*mb.pd0.label: Gra
 XShisen*mb.pd1.label: Operacje
 XShisen*mb.pd2.label: Konfiguracja
-XShisen*mb.pd3.label: Picture
+XShisen*mb.pd3.label: Kafle
 XShisen*pd0*button_0.label: Start           Ctrl-D
 XShisen*pd0*button_1.label: Restart         Ctrl-R
 XShisen*pd0*button_2.label: Pauza           Ctrl-P
@@ -131,10 +131,10 @@
 XShisen*pd2*button_0.label: Normalne  (18x 8)
 XShisen*pd2*button_1.label: Duøe      (24x12)
 XShisen*pd2*button_2.label: Ogromne   (36x16)
-XShisen*pd2*button_3.label: Click Trial
+XShisen*pd2*button_3.label: Szybkie klikanie
 XShisen*pd2*button_4.label: Grawitacja
-XShisen*pd3*button_0.label: Picture#1
-XShisen*pd3*button_1.label: Picture#2
+XShisen*pd3*button_0.label: Zestaw 1
+XShisen*pd3*button_1.label: Zestaw 2
 XShisen*sform*translations: #replace <Btn1Down>: PickupPiece()\n\
         <Btn2Down>(2): menu(0,0)\n\
         <Btn3Down>: CancelPiece()\n\
diff -Nur .orig/lang/XShisen.ad.ja_JP.JIS7 /lang/XShisen.ad.ja_JP.JIS7
--- lang/XShisen.ad.ja_JP.JIS7.orig
+++ lang/XShisen.ad.ja_JP.JIS7
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: $B;M@n>J(B for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: $B;M@n>J(B for X11 Ver 1.50\n\n\
  $B;M@n>J(B for X11     by $B>.NS(B $B@56=(B <masaoki@techfirm.co.jp>\n\
    $B;M@n>J(B for Win   by $B>.Ln(B $B?8Fs(B <ono@urban.ne.jp>\n\
    $BGW%G%6%$%s(B       by $B=);3(B $BN4G7(B <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja_JP.ujis /lang/XShisen.ad.ja_JP.ujis
--- lang/XShisen.ad.ja_JP.ujis.orig
+++ lang/XShisen.ad.ja_JP.ujis
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: ªÕ¿Óæ  for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: ªÕ¿Óæ  for X11 Ver 1.50\n\n\
  ªÕ¿Óæ  for X11     by æÆŒ” ¿µ∂Ω <masaoki@techfirm.co.jp>\n\
    ªÕ¿Óæ  for Win   by æÆÃÓ ø∏∆Û <ono@urban.ne.jp>\n\
    «◊•«•∂•§•Û       by Ω©ª≥ Œ¥«∑ <taco3@ybb.ne.jp>
diff -Nur .orig/lang/XShisen.ad.ja_JP.PCK /lang/XShisen.ad.ja_JP.PCK
--- lang/XShisen.ad.ja_JP.PCK.orig
+++ lang/XShisen.ad.ja_JP.PCK
@@ -1,7 +1,7 @@
 ! Common resources
 XShisen*connectLineColor: blue
 XShisen*connectLineTime: 1000
-XShisen*aboutString: élêÏè» for X11 Ver XSHISEN_VERSION\n\n\
+XShisen*aboutString: élêÏè» for X11 Ver 1.50\n\n\
  élêÏè» for X11     by è¨ó— ê≥ãª <masaoki@techfirm.co.jp>\n\
    élêÏè» for Win   by è¨ñÏ êWìÒ <ono@urban.ne.jp>\n\
    îvÉfÉUÉCÉì       by èHéR ó≤îV <taco3@ybb.ne.jp>
