diff -Nur man/xshisen.6 man/xshisen.6
--- man/xshisen.6.old	2002-07-10 11:00:41.000000000 +0800
+++ man/xshisen.6	2006-02-21 16:29:10.000000000 +0800
@@ -137,7 +137,7 @@
 Button binding setting for your mouse.  Default is
 .NF
 
-      <Btn1Down>: #replace PickupPiece() \n <Btn3Down>: CancelPiece()
+      <Btn1Down>: #replace PickupPiece() \\n <Btn3Down>: CancelPiece()
 
 .FI
 
diff -Nur man/xshisen.6.ja man/xshisen.6.ja
--- man/xshisen.6.ja.old   2002-07-10 11:00:29.000000000 +0800
+++ man/xshisen.6.ja	   2006-02-21 16:29:32.000000000 +0800
@@ -143,7 +143,7 @@
 マウスのボタンバインディングを変更します。デフォルトは
 .NF
 
-      <Btn1Down>: #replace PickupPiece() \n <Btn3Down>: CancelPiece()
+      <Btn1Down>: #replace PickupPiece() \\n <Btn3Down>: CancelPiece()
 
 .FI
 です。
diff -Nur man/xshisen.6.ja_JP man/xshisen.6.ja_JP
--- man/xshisen.6.ja_JP.old	2002-07-10 11:00:29.000000000 +0800
+++ man/xshisen.6.ja_JP		2006-02-21 16:29:32.000000000 +0800
@@ -143,7 +143,7 @@
 マウスのボタンバインディングを変更します。デフォルトは
 .NF
 
-      <Btn1Down>: #replace PickupPiece() \n <Btn3Down>: CancelPiece()
+      <Btn1Down>: #replace PickupPiece() \\n <Btn3Down>: CancelPiece()
 
 .FI
 です。
diff -Nur man/xshisen.6.ja_JP.eucJP man/xshisen.6.ja_JP.eucJP
--- man/xshisen.6.ja_JP.eucJP.old	2002-07-10 11:00:29.000000000 +0800
+++ man/xshisen.6.ja_JP.eucJP		2006-02-21 16:29:32.000000000 +0800
@@ -143,7 +143,7 @@
 マウスのボタンバインディングを変更します。デフォルトは
 .NF
 
-      <Btn1Down>: #replace PickupPiece() \n <Btn3Down>: CancelPiece()
+      <Btn1Down>: #replace PickupPiece() \\n <Btn3Down>: CancelPiece()
 
 .FI
 です。
diff -Nur man/xshisen.6.ja_JP.ujis man/xshisen.6.ja_JP.ujis
--- man/xshisen.6.ja_JP.ujis.old	2002-07-10 11:00:29.000000000 +0800
+++ man/xshisen.6.ja_JP.ujis		2006-02-21 16:29:32.000000000 +0800
@@ -143,7 +143,7 @@
 マウスのボタンバインディングを変更します。デフォルトは
 .NF
 
-      <Btn1Down>: #replace PickupPiece() \n <Btn3Down>: CancelPiece()
+      <Btn1Down>: #replace PickupPiece() \\n <Btn3Down>: CancelPiece()
 
 .FI
 です。
