--- analyzer/soxprovider.h.orig	2008-03-08 19:53:58.000000000 +1100
+++ analyzer/soxprovider.h	2008-03-08 19:54:18.000000000 +1100
@@ -18,7 +18,7 @@
     if (extension == "mp3" || extension == "ogg")
         command << "-t ." << extension << " ";
     command << "\'" << epath << "\' ";
-    command << "-t .raw -w -c 1 -r " << samplerate
+    command << "-t .raw -4 -c 1 -r " << samplerate
         << (sign ? " -s" : " -u") << " -";
     LOG(INFO) << "Executing: " << command.str() << endl;
     return popen(command.str().c_str(), "r");
