--- convert.conf.old	2009-06-26 19:51:50.000000000 +0100
+++ convert.conf	2009-07-10 08:30:31.000000000 +0100
@@ -75,15 +75,15 @@
 
 mp4 mp3 * *
 	# FB:{BITRATE=-B %B}
-	[faad] -q -w -f 1 $FILE$ | [lame] --silent -q $QUALITY$ $BITRATE - -
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [lame] --silent -q $QUALITY$ $BITRATE - -
 
 aac mp3 * *
 	# IFB:{BITRATE=-B %B}
-	[faad] -q -w -f 1 $FILE$ | [lame] --silent -q $QUALITY$ $BITRATE - -
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [lame] --silent -q $QUALITY$ $BITRATE - -
 
 mov mp3 * *
-	# FRB:{BITRATE=-B %B}D:{RESAMPLE=--resample %D}
-	[mov123] $FILE$ | [lame] -s 44100 --silent -q $QUALITY$ $RESAMPLE$ -v $BITRATE$ -r --big-endian - -
+	# FB:{BITRATE=-B %B}D:{RESAMPLE=--resample %D}
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [lame] -s 44100 --silent -q $QUALITY$ $RESAMPLE$ -v $BITRATE$ -r --big-endian - -
 
 alc mp3 * *
 	# FB:{BITRATE=-B %B}D:{RESAMPLE=--resample %D}
@@ -167,15 +167,15 @@
 
 mp4 wav * *
 	# F
-	[faad] -q -w -f 2 -b 1 -s 44100 $FILE$
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1
 
 aac wav * *
 	# IF
-	[faad] -q -w -f 2 -b 1 -s 44100 $FILE$
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1
 
-mov aif * *
-	# FR
-	[mov123] $FILE$
+mov wav * *
+	# F
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1
 
 wma wma * *
 	-
@@ -214,15 +214,15 @@
 
 mp4 flc * *
 	# F
-	[faad] -q -w -f 1 $FILE$ | [flac] -cs --totally-silent --compression-level-0 --ignore-chunk-sizes -
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [flac] -cs --totally-silent --compression-level-0 --ignore-chunk-sizes -
 
 aac flc * *
 	# IF
-	[faad] -q -w -f 1 $FILE$ | [flac] -cs --totally-silent --compression-level-0 --ignore-chunk-sizes -
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [flac] -cs --totally-silent --compression-level-0 --ignore-chunk-sizes -
 
 mov flc * *
-	# FR
-	[mov123] $FILE$ | [flac] -cs --totally-silent --compression-level-0 --endian big --sign signed --channels 2 --bps 16 --sample-rate 44100 -
+	# F
+	[mplayer] -novideo -ao pcm:file=/dev/fd/4 $FILE$ 4>&1 1>/dev/null 2>&1 | [flac] -cs --totally-silent --compression-level-0 --endian little --sign signed --channels 2 --bps 16 --sample-rate 44100 -
 
 alc flc * *
 	# F
