--- Slim/Formats/XML.pm.old	2009-01-17 21:27:29.000000000 +0000
+++ Slim/Formats/XML.pm	2009-01-17 21:40:14.000000000 +0000
@@ -13,7 +13,7 @@
 use strict;
 use File::Slurp;
 use HTML::Entities;
-use JSON::XS qw(from_json);
+use JSON::XS qw(decode_json);
 use Scalar::Util qw(weaken);
 use URI::Escape qw(uri_escape);
 use XML::Simple;
@@ -288,7 +288,7 @@
 	my $xml;
 	
 	if ( $type =~ /json/ ) {
-		$xml = from_json($$content);
+		$xml = decode_json($$content);
 	}
 	else {
 		$xml = xmlToHash($content);
