diff -ur adeona/common/options.c adeona.new/common/options.c
--- adeona/common/options.c	2008-07-14 11:30:54.000000000 +1000
+++ adeona.new/common/options.c	2008-08-04 19:27:05.938902141 +1000
@@ -36,16 +36,16 @@
 #define OPTION_END                "END_OPTIONS"
 
 //NOTE: all strings must be less than 1023 bytes.
-#define DEFAULT_RESOURCE_PATH      "resources/"
+#define DEFAULT_RESOURCE_PATH      "/var/lib/adeona/resources"
 #define	DEFAULT_LOG_FILENAME       "adeona.log" 
 #define DEFAULT_SERVER_PORT        5852
 #define DEFAULT_CTXT_QUEUE_SIZE    1
 #define DEFAULT_SCHED_LAMBDA       0.0006
 #define DEFAULT_SERVER_NAME        "opendht.nyuld.net" 
 #define DEFAULT_NUM_DHTKEYS_TOTRY  2
-#define DEFAULT_GATEWAYS_FILENAME  "resources/gateways.adeona"
+#define DEFAULT_GATEWAYS_FILENAME  "/var/lib/adeona/resources/gateways.adeona"
 #define DEFAULT_USE_TRACEROUTES    1
-#define	DEFAULT_LANDMARKS_FILENAME "resources/landmarks.adeona"
+#define	DEFAULT_LANDMARKS_FILENAME "/var/lib/adeona/resources/landmarks.adeona"
 #define	DEFAULT_MAX_SLEEPTIME      300 // 5 minutes
 #define DEFAULT_USE_WEBCAMIMG      0
 #define DEFAULT_WEBCAMPROG_PATH    "resources/isightcapture.exe"
diff -ur adeona/location/location_test.c adeona.new/location/location_test.c
--- adeona/location/location_test.c	2008-07-14 11:30:55.000000000 +1000
+++ adeona.new/location/location_test.c	2008-08-04 19:26:09.182902129 +1000
@@ -40,7 +40,7 @@
     uchar* trdata = NULL;
     int len = 0;
     
-    landmarks_load( "resources/landmarks.adeona" );
+    landmarks_load( "/var/lib/adeona/resources/landmarks.adeona" );
     
     get_traceroutes( &trdata, &len );
     print_traceroutes( stdout, trdata, len, NORMAL );
@@ -58,7 +58,7 @@
     location_t loc;
 
     location_alloc( &loc );
-    landmarks_load( "resources/landmarks.adeona" );
+    landmarks_load( "/var/lib/adeona/resources/landmarks.adeona" );
     assert( landmarks_isloaded() == TRUE );
 
     location_get_data( &loc );
diff -ur adeona/Makefile adeona.new/Makefile
--- adeona/Makefile	2008-07-14 11:30:55.000000000 +1000
+++ adeona.new/Makefile	2008-08-04 19:28:52.120904206 +1000
@@ -35,27 +35,27 @@
 #
 ## Build options:
 #
-ADEONA_BUILD_MODE := "debug"
-ifeq "$(origin RELEASE)" "command line"
-ifeq "$(RELEASE)" "1"
-	ADEONA_BUILD_MODE := "release"
-endif
-endif
+ADEONA_BUILD_MODE := "release"
+# ifeq "$(origin RELEASE)" "command line"
+# ifeq "$(RELEASE)" "1"
+# 	ADEONA_BUILD_MODE := "release"
+# endif
+# endif
 
 
-CFLAGS := -Wall
+CFLAGS := -Wall -D_ADEONA_BUILD_RELEASE_
 
 ifeq ($(PLATFORM), "MacOS")
 CFLAGS += -mmacosx-version-min=$(MACOSVERSION) -arch ppc -arch i386
 endif
 
-ifeq ($(ADEONA_BUILD_MODE), "debug")
-CFLAGS += -D_ADEONA_BUILD_DEBUG_
-CFLAGS += -ggdb
-else
-CFLAGS += -D_ADEONA_BUILD_RELEASE_
-CFLAGS += -O2
-endif
+# ifeq ($(ADEONA_BUILD_MODE), "debug")
+# CFLAGS += -D_ADEONA_BUILD_DEBUG_
+# CFLAGS += -ggdb
+# else
+# CFLAGS += -D_ADEONA_BUILD_RELEASE_ -DDEFAULT_RESOURCE_PATH="/var/lib/adeona/resources"
+# CFLAGS += -O2
+# endif
 
 #
 ## Setup paths:
